orchagent/routeorch.cpp

```diff
@@ -1600,6 +1690,13 @@ bool RouteOrch::addRoutePost(const RouteBulkContext& ctx, const NextHopGroupKey
         /* Increase the ref_count for the next hop (group) entry */
         increaseNextHopRefCount(nextHops);
 
+        if (nextHops.getSize() == 1)
+        {
+            RouteKey r_key = { vrf_id, ipPrefix };
+            auto nexthop = NextHopKey(nextHops.to_string());
+            addNextHopRoute(nexthop, r_key);
+        }
```

Comment:
what is the purpose of add the <nexthop, r_key> ? is this used for track the nexthop change, so that we can find the route to be changed?

however, what about if there is a T1 route has only single nexthop?

Reply:
also, what is the implication if we have this nexthop route table on T1? feel it is totally unnecessary.

---
In reply to: [568514575](https://github.com/Azure/sonic-swss/pull/1615#discussion_r568514575) [](ancestors = 568514575)

Reply:
This is a common data structure for holding Single NH -> Routes. It includes both T1 and Mux routes. I think it is hard to differentiate with T1 routes. Do you have a suggestion?

Reply:
first, if the muxorch is not enabled, we should not enable this nexthop -> route tracking.

Reply:
isn't true we will know which port has muxcable, and only track nexthop if it is pointing that port?

Reply:
The nexthop map is required for evpn case. Also to clarify, this does not contain T1 routes since in most cases its ECMP. This datastructure is only if a route has single nexthop all the time.